integration_test/test_compute_metrics.py

```diff
@@ -0,0 +1,130 @@
+  report_compute = compute_classification_report(y_true, y_pred)
+  report_sklearn = classification_report(y_true, y_pred, output_dict=True, zero_division=0)
+  assert np.abs(report_compute['weighted avg']['precision'] - report_sklearn['weighted avg']['precision']) < 1e-8
+  assert np.abs(report_compute['weighted avg']['recall'] - report_sklearn['weighted avg']['recall']) < 1e-8
+  assert np.abs(report_compute['weighted avg']['f1-score'] - report_sklearn['weighted avg']['f1-score']) < 1e-8
+  assert np.abs(compute_accuracy(y_true, y_pred) - accuracy_score(y_true, y_pred)) < 1e-8
+
+
+def assert_regression_metrics(y_true, y_pred):
+  assert np.abs(mean_absolute_error(y_true, y_pred) - compute_mae(y_true, y_pred)) < 1e-8
+  assert np.abs(mean_squared_error(y_true, y_pred) - compute_mse(y_true, y_pred)) < 1e-8
+
+
+iris = datasets.load_iris()
+X = iris.data
+y = iris.target
+num_class = 3
+X_train, X_test, Y_train, Y_test = train_test_split(X, y, test_size=0.3)
+D_train = xgb.DMatrix(X_train, label=Y_train)
+D_test = xgb.DMatrix(X_test, label=Y_test)
+
+
+class TestComputeXGBMetrics(object):
+  def test_classification_metrics(self):
+    params = {
+      'eta': 0.3,
+      'num_class': num_class,
+      'objective': 'multi:softmax'
+    }
+    xgb_params = {
+      'params': params,
+      'dtrain': D_train,
+      'evals': [(D_test, 'test0')],
+    }
+    ctx = sigopt.xgboost.run(**xgb_params)
+    bst = ctx.model
+
+    # compute sklearn-based metrics
+    preds = bst.predict(D_test)
+    xgb_preds = np.round(preds)
+    xgb_true = Y_test
+    report_sklearn = classification_report(xgb_true, xgb_preds, output_dict=True, zero_division=0)
+    accuracy_sklearn = accuracy_score(xgb_true, xgb_preds)
+
+    # compare metrics
+    run = sigopt.get_run(ctx.run.id)
+    assert np.abs(run.values['test0-precision'].value - report_sklearn['weighted avg']['precision']) < 1e-8
+    assert np.abs(run.values['test0-recall'].value - report_sklearn['weighted avg']['recall']) < 1e-8
+    assert np.abs(run.values['test0-F1'].value - report_sklearn['weighted avg']['f1-score']) < 1e-8
+    assert np.abs(run.values['test0-accuracy'].value - accuracy_sklearn) < 1e-8
+
+  def test_regression_metrics(self):
+    params = {
+      'eta': 0.3,
+      'objective': 'reg:squarederror'
+    }
+    xgb_params = {
+      'params': params,
+      'dtrain': D_train,
+      'evals': [(D_test, 'test0')],
+    }
+    ctx = sigopt.xgboost.run(**xgb_params)
+    bst = ctx.model
+
+    # compute sklearn-based metrics
+    preds = bst.predict(D_test)
+    xgb_preds = np.round(preds)
+    xgb_true = Y_test
+    mae = mean_absolute_error(xgb_true, xgb_preds)
+    mse = mean_squared_error(xgb_true, xgb_preds)
+
+    # compare metrics
+    run = sigopt.get_run(ctx.run.id)
+    assert np.abs(run.values['test0-mean absolute error'].value - mae) < (1e-8 / mae)
+    assert np.abs(run.values['test0-mean squared error'].value - mse) < (1e-8 / mae)
```

Comment:
move all these into `integration_test`

Reply:
Okay I think there was a misunderstanding. So right now we have three sets of tests corresponding to three classes: 

- TestComputeXGBMetrics (check if xgb.run records correct metrics)
- TestComputeSyntheticMetrics (checks a variety of metric computations using synthetic stuff)
- TestLogMetrics (just checks logging)

Which of these ought to be in integration_test/ and which of these ought to be in test/ ?